.github/workflows/backend.yml

```
@@ -44,31 +45,44 @@ jobs:
         cache-dependency-path: ui/pnpm-lock.yaml
     - name: Install pip
       run: python -m pip install build requests virtualenv
-    - name: Install Gradio
+    - name: venv activate Linux
+      if: ${{ matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest' }}
```

We can remove the macos:
```suggestion
      if: ${{ matrix.os == 'ubuntu-latest'}}
```